workloads/imagenet/imagenet_pytorch/workload.py

```diff
@@ -0,0 +1,235 @@
+"""Resnet workload implemented in PyTorch."""
+import contextlib
+import os
+from collections import OrderedDict
+from typing import Tuple
+
+import torch
+import torch.nn.functional as F
+from torchvision import transforms
+from torchvision.datasets import ImageNet
+from torchvision.datasets.folder import ImageFolder
+
+import spec
+import random_utils as prng
+from workloads.imagenet.workload import ImagenetWorkload
+from workloads.imagenet.imagenet_pytorch.resnet import resnet50
+from workloads.imagenet.imagenet_pytorch.utils import fast_collate, cycle
+
+
+DEVICE='cuda' if torch.cuda.is_available() else 'cpu'
+
+
+class ImagenetWorkload(ImagenetWorkload):
+
+  def __init__(self):
+    super().__init__()
+    self.dataset = 'imagenet2012'
+
+  """ data augmentation settings """
+  @property
+  def scale_ratio_range(self):
+    return (0.08, 1.0)
+
+  @property
+  def aspect_ratio_range(self):
+    return (0.75, 4.0 / 3.0)
+
+  @property
+  def center_crop_size(self):
+    return 224
+
+  @property
+  def resize_size(self):
+    return 256
+
+  @property
+  def num_train_examples(self):
+    return 1281167
+
+  @property
+  def num_eval_examples(self):
+    return 50000
+
+  @property
+  def param_shapes(self):
+    """
+    TODO: return shape tuples from model as a tree
+    """
+    raise NotImplementedError
```

Comment:
@fsschneider and me discussed that it would be nice to use the `param_shapes` property to test the equivalence of the Jax and PyTorch model implementation (obviously it doesn't test for all potential differenes, such as different initializations). So we would simply have a pytest which checks if the `param_shapes` property of the the Jax and PyTorch workload instance matches. I created an issue for it (#53) and will take care of it in a separate PR (the PyTorch MNIST workload also doesn't have this property implemented yet).